Src/GoogleApis.Auth/OAuth2/Flows/GoogleAuthorizationCodeFlow.cs

```diff
@@ -72,12 +78,20 @@ public override AuthorizationCodeRequestUrl CreateAuthorizationCodeRequest(strin
             await DeleteTokenAsync(userId, taskCancellationToken);
         }
 
+        public override bool ShouldForceTokenRetrieval() 
+        {
+            return includeGrantedScopes.HasValue && includeGrantedScopes.Value == true;

```

Comment:
Done. Sorry, I tend to write code almost as verbosely as I do when I write everything else. I'm sure you've noticed =D